Jenkinsfile

```diff
@@ -1,216 +1,116 @@
+                            } catch(e) {
+                                // Die right here
+                                error stageStatus
                             }
-                            catch(e) {
-                                build_ok = false
-                                errors.add(env.STAGE_STATUS)
-                                println e.toString()
-                            }
-                            sh 'source deactivate'
 
-                            try{
-                                sh """
-                                    source activate ${condaPath}/envs/isis 
-                                    ctest -R _module_ -j4 -VV
-                                """
+                            // Unit tests
+                            stageStatus = "Running unit tests on ${label}"
+                            try {
+                                loginShell "ctest -R _unit_ -j${NUM_CORES} -VV"
+                            } catch(e) {
+                                errors.add(stageStatus)
+                                currentBuild.result = 'FAILURE'
                             }
-                            catch(e) {
-                                build_ok = false
-                                errors.add(env.STAGE_STATUS)
-                                println e.toString()
+
+                            // App tests
+                            stageStatus = "Running app tests on ${label}"
+                            try {
+                                loginShell "ctest -R _app_ -j${NUM_CORES} -VV"
+                            } catch(e) {
+                                errors.add(stageStatus)
+                                currentBuild.result = 'FAILURE'
                             }
-                            sh 'source deactivate'
 
-                            try{
-                                env.STAGE_STATUS = "Running gtests on ${label}"
-                                sh """
-                                    source activate ${condaPath}/envs/isis
-                                    ctest -R "." -E "(_app_|_unit_|_module_)" -j4 -VV
-                                """
+                            try {
+                                loginShell "ctest -R _module_ -j${NUM_CORES} -VV"
+                            } catch(e) {
+                                errors.add(stageStatus)
+                                currentBuild.result = 'FAILURE'
                             }
-                            catch(e) {
-                                build_ok = false
-                                errors.add(env.STAGE_STATUS)
-                                println e.toString()
+
+                            // Gtests
+                            stageStatus = "Running gtests on ${label}"
+                            try {
+                                loginShell "ctest -R '.' -E '(_app_|_unit_|_module_)' -j${NUM_CORES} -VV"
```

Comment:
any significant difference between loginshell and sh? I would assume login shell to change depending on the default login shell in the container/mac machine rather than always using sh? In which case is there any reason we wouldn't want to force bash?

Reply:
loginShell forces bash and adds the '-l' switch, which makes sure commands like 'conda activate' work: https://code.usgs.gov/astrogeology/jenkinslib/-/blob/master/vars/loginShell.groovy